### PR TITLE
BUG: fix apperance stream options for forms

### PR DESCRIPTION
**Pull request by fidoriel**
add apperance stream option to update_page_form_field_values. If true it generates apperance streams but only for one line.
There are also issues with non-ascii characters not showing.
This is a fix for #355.

### CODE REVIEW

File: PyPDF2/_writer.py [hunk 1]
```diff
@@ -620,6 +623,11 @@ def update_page_form_field_values(
         :param int flags: An integer (0 to 7). The first bit sets ReadOnly, the
             second bit sets Required, the third bit sets NoExport. See
             PDF Reference Table 8.70 for details.
+        :param bool generate_apperance_streams: Some PDF reader can generate a apperance
+            atream when opening the PDF. Others, often simple one (previews etc.), do not.
```
Comment (MartinThoma) [inline PyPDF2/_writer.py:627]:
```suggestion
            stream when opening the PDF. Others, often simple one (previews etc.), do not.
```

**Comment by MartinThoma** [comment]
Thank you for taking care of it :heart:  It would be amazing if we could finally close this long-standing issue!

I'll try to test it on Saturday. In the mean time, would you mind to run `black .` (installable via `pip install black`) to get rid of the flake8 issues?

**Comment by brzGatsu** [comment]
Hoping for a quick integration and release. With this change I will start using PyPdf!

**Comment by fidoriel** [comment]
I will check the changes in https://github.com/Vigrond/PyPDF2/tree/appearance-stream made to my PR and integrate them. Maby we will find a solution for multi-line word px calculation and the encoding.

**Comment by MartinThoma** [comment]
How are things going @fidoriel ?

**Comment by fidoriel** [comment]
Since my last message I did not invest time into this isssue. Not using external dependencies makes it very hard to calculate text px based on the font.

**Comment by MartinThoma** [comment]
I understand that. Is there a specific external dependency that would help a lot?

@pubpub-zz Did you calculate the text px size anywhere already?

**Comment by fidoriel** [comment]
I think https://github.com/python-pillow/Pillow can do the job

**Comment by pubpub-zz** [comment]
If you mean the Height, Yes : the function is available, for horizontal no.
@fidoriel,
It is a little easier and complex at the same time : widths are already provided in the fonts
There is still some work to extract the data and compute for each line,

**Comment by brzGatsu** [comment]
Sorry if this is a stupid question, but why is text measuring important? I thought the PDF reader takes care of form rendering? Asking because I'd rather have the form filling finally working than wait for edge-case fixes.

**Comment by MartinThoma** [comment]
What is the state of this PR? Does it improve pypdf?

What is missing to get it merged? (besides resolving the merge conflict)?

**Comment by fidoriel** [comment]
My comany has decided to not use pypdf in our Project.
What is needed:
- The calculation of the Font size/length and sentence breaks
- Encode Unicode correctly
- There are 2 forks of my brach wich might also help as far as I remember
- Testing that this really fix the issue on all PDF reader

**Comment by pubpub-zz** [comment]
> My comany has decided to not use pypdf in our Project. What is needed:
> 
>     * The calculation of the Font size/length and sentence breaks
> 
>     * Encode Unicode correctly
> 
>     * There are 2 forks of my brach wich might also help as far as I remember
> 
>     * Testing that this really fix the issue on all PDF reader

@fidoriel 
You should create some PR from your forks.

**Comment by MartinThoma** [comment]
> My comany has decided to not use pypdf in our Project.

I'm curious: Can you let me know (maybe in private; info@martin-thoma.de) what you use instead?

**Comment by MartinThoma** [comment]
I'm closing this PR now as it will likely never get merged. I'm sorry.

@pubpub-zz Maybe here are some ideas that would be useful for our current code base. If you do, please give credit in the commit so that we don't loose that information.

**Comment by pubpub-zz** [comment]
I've missed it a while ago (sorry about that @fidoriel) but it is another way to fill up forms.
The current issues I'm facing are further than this (how to introduce a fulll range true type fonts to allow filling up with Arabic,...
@fidoriel If you have any ideas about It will be welcomed

**Pull request closed without merging**